include/odp/api/spec/traffic_mngr.h

```diff
@@ -638,6 +668,41 @@ void odp_tm_egress_init(odp_tm_egress_t *egress);
 int odp_tm_capabilities(odp_tm_capabilities_t capabilities[],
 			uint32_t              capabilities_size);
 
+/** Query All TM Capabilities specific to an egress
+ *
+ * The odp_tm_egress_capabilities() function can be used to obtain the complete
+ * set of TM limits supported by this implementation for a given egress.
+ * The reason that this returns a SET of capabilities and not just one, is
+ * because it is expected that many HW based implementations may have one set
+ * of limits for the HW and also support a SW TM implementation with a
+ * (presumably larger) different set of limits. There are also cases where
+ * there could be more than SW implementation (one supporting say tens of
+ * thousands of tm_queues and a variant supporting tens of millions of
+ * tm_queues). This is needed as platforms might have different constraints
+ * for different egress type and there is no way to know them using
+ * odp_tm_capabilities() provided capabilities. The caller passes in an array
+ * of odp_tm_capabilities_t records and the maximum number of such records to
+ * output. If number of such records implementation supports is larger than
+ * caller requested number, then only caller requested number of records are
+ * written and return value is max number of records implementation supports.
+ * Caller then may again call with larger number of records to be returned.
+ *
+ * @param[out] capabilities      An array of odp_tm_capabilities_t records to
+ *                               be filled in.
+ * @param      capabilities_size The number of odp_tm_capabilities_t records
+ *                               in the capabilities array.
+ * @param      egress            Only capabilities compatible with this egress
+ *                               are returned.
+ * @return                       Returns < 0 upon failure. Returns N > 0,
+ *                               where N is the maximum number of different
+ *                               odp_tm_capabilities_t records that the
+ *                               implementations supports. *NOTE* that this
+ *                               number can be > capabilities_size!
+ */
+int odp_tm_egress_capabilities(odp_tm_capabilities_t capabilities[],
```

Comment:
Now that there is this egress specific capability call, do we need odp_tm_capabilities() anymore? If it stays in the API, what does it return? If it returns something that is not supported by all egresses, then it cannot be relied on and it appears rather useless. So if it stays, maybe it should return the intersection of the capabilities of all egresses (i.e. the capabilities that are supported by all egresses), but even then I wonder if it would be of much use to an application.

Reply:
We are fine to deprecate older call.

Reply:
I think the behaviour of the non-egress specific capability function should be clarified (i.e. it returns capabilities that are valid for all egresses) even if the function gets deprecated.